lib/core/note.dart

```diff
@@ -359,7 +370,7 @@ class Note implements File {
   bool get shouldRebuildPath {
     if (fileFormat != NoteFileFormat.Markdown) return false;
 
-    return parent.config.fileNameFormat == NoteFileNameFormat.FromTitle;
```

Comment:
This is the source of https://github.com/GitJournal/GitJournal/issues/654.

The filename needs to be rebuilt whenever the file name format includes the title, which includes other cases than the plain "title" option.